src/internal.c

```diff
@@ -5818,6 +5829,15 @@ int DhGenKeyPair(WOLFSSL* ssl, DhKey* dhKey,
 
     WOLFSSL_ENTER("DhGenKeyPair");
 
+#if defined(HAVE_PK_CALLBACKS)
+    if (ssl && ssl->ctx && ssl->ctx->DhGenerateKeyPairCb) {
+        ret = ssl->ctx->DhGenerateKeyPairCb(dhKey, ssl->rng, priv, privSz,
+                                            pub, pubSz);
+        if (ret != NOT_COMPILED_IN)
+            return ret;
+    }
```

Comment:
Now that async goes first, the current form has the desired effect.  That is, if `ssl->ctx->DhGenerateKeyPairCb()` returns `NOT_COMPILED_IN` , it will fallback to software.

Reply:
Ah!!  sorry, I got mixed up because I saw `else if ....`  in the first message. I know what to do now.